ql/experimental/finitedifferences/fdmvppstepcondition.cpp

```diff
@@ -64,27 +64,24 @@ namespace QuantLib {
 
 
     void FdmVPPStepCondition::applyTo(Array& a, Time t) const {
-        ext::shared_ptr<FdmLinearOpLayout> layout = mesher_->layout();
```

Comment:
In cases like this, something like 

    const auto& layout = *mesher_->layout();

would still prevent copies and also avoid repeating `mesher_->layout()` below &mdash; what do you think?

Reply:
Yes, that is true. In a lot of the files I touched there were already other instances of `mesher->layout()` in the same scope, but in cases like this a local reference could result in better readability. I can make that change if you think it's nicer.

Reply:
If you have time, of course.

Reply:
After taking another look, I don't think it will be easy to make this change in a non-arbitrary way. Plus, I prefer to omit the local variable as a matter of personal style, so if it's ok with you I think I'll stick with the original commit.

Reply:
Ok, no problem.